### ISSUE TITLE
Show BOT flag

### ISSUE THREAD

**Issue by nivex** [matterhorn-chat/matterhorn#494]
In the standard web interface, some posts have the username suffixed with "BOT" to indicate the response came from an integration. It would be nice if Matterhorn would render this as well.

**Comment by jtdaugherty** [comment]
From my discussion with in the `pre-release` Matterhorn channel just now:
```
[11:28] jtdaugherty: @harrison in the web client, usernames with the "BOT" suffix seem to
  be those whose messages are delivered by a webhook. Is that the correct association?
  In other words, does "BOT" in the username mean the from_webhook post prop was set
  to True?
[11:30] │ harrison: Prior to the 5.10 that's coming out in a couple weeks, that's correct
[11:30] │ harrison: In 5.10, there'll also be bot users as well where all of their posts
  are marked with that same BOT label
[11:30] │ jtdaugherty: How will those users be identified by their metadata?
[11:31] │ harrison: They'll have a user.is_bot field set to true in the json
[11:31] │ jtdaugherty: Okay, thanks!
```

**Comment by jtdaugherty** [comment]
@nivex Matterhorn already has this behavior but perhaps by not looking at the post prop I mentioned above, we're missing a case. Can you say a bit more about the kind of situation where you're seeing a user without the suffix that should get one? I'm updating our logic to consider the post prop but it would still help to know how else this situation can arise other than the one we already support.

**Comment by nivex** [comment]
I have some slash commands configured very minimally, so they post as the user that invoked it.

Here is a screenshot of the line in question (with preceding lines for a little context) from the web interface:
![mattermost-web-bot](https://user-images.githubusercontent.com/3833404/55428210-9e20aa00-5556-11e9-8108-3ffd7a0f4463.png)

And the same lines as seen from Matterhorn:
![matterhorn-nobot](https://user-images.githubusercontent.com/3833404/55428283-c5777700-5556-11e9-95c0-092f7a9ce70c.png)

**Comment by jtdaugherty** [comment]
Would it be possible for you to open your browser development console and look for the network response to the `posts?page=...` API request that the client makes when you enter that channel, and provide the JSON for one of those posts?

**Comment by nivex** [comment]
I see the `from_webhook` property you're talking about:
```json
"t6njkkabrpn9drwx5egenjqxaa":{"id":"t6njkkabrpn9drwx5egenjqxaa","create_at":1554149195754,"update_at":1554149195754,"edit_at":0,"delete_at":0,"is_pinned":false,"user_id":"nnfdsucqgjnxdphfgrcydne3jc","channel_id":"bd77aq3dniyqd8f7z4d1yxdsnw","root_id":"","parent_id":"","original_id":"","message":"N8VNR, KEVIN J OTTE, HILLSBOROUGH, NC 27278, EXTRA - https://callook.info/n8vnr","type":"","props":{"from_webhook":"true"},"hashtags":"","pending_post_id":"","metadata":{}}
```

**Comment by jtdaugherty** [comment]
Ah! That explains it. Thanks! That message has `from_webhook` set (which we are ignoring at the moment) but *not* the other fields that we *do* look at. So the fix I'm working on will be adequate.

**Comment by jtdaugherty** [comment]
This is now fixed in `develop` and will go out in the next major release. Thank you for the report!

**Comment by jtdaugherty** [comment]
This is now released in 50200.3.0.
